src/rules/jsxCurlySpacingRule.ts

```diff
@@ -125,36 +122,76 @@ class JsxCurlySpacingWalker extends Lint.RuleWalker {
         const nodeWidth = node.getWidth();
 
         if (this.hasOption(OPTION_ALWAYS)) {
-            if (!isSpaceBetweenTokens(firstToken, secondToken)) {
+            if (!this.getDeleteFixForSpaceBetweenTokens(firstToken, secondToken)) {
```

Comment:
this is now a little hard to read because "getting" the fix and checking if there is an issue are now conflated into one function call... I prefer how you did it below in the other branches, where you try to get the fix first and then check if it exists. also please avoid implicit type coercion.